src/ai/defaultai.cc

```diff
@@ -862,46 +864,38 @@ void DefaultAI::late_initialization() {
 			// Some important buildings are identified
 			if (prod.input_wares().empty() && !prod.output_ware_types().empty() &&
 			    prod.created_immovables().empty() && !prod.collected_immovables().empty()) {
-				bool produces_construction_material = false;
-				bool produces_non_construction_material = false;
-				for (Widelands::DescriptionIndex output_idx : prod.output_ware_types()) {
-					if (tribe_->is_construction_material(output_idx)) {
-						produces_construction_material = true;
-					} else {
-						produces_non_construction_material = true;
-					}
-				}
-				if (produces_construction_material) {
-					// TODO(GunChleoc): We should lose the hard distinction between quarry and
-					// lumberjack, so that a building can be both
-					if (prod.supported_by_productionsites().empty()) {
+				// TODO(hessenfarmer): hardcoded strings should be parsed from a definition file
+				for (const auto& attribute : prod.collected_attributes()) {
+					if (attribute.second == Widelands::MapObjectDescr::get_attribute_id("rocks")) {
```

Comment:
changed the detection of some special productionsites to check for the collected immovables, as the former logic was based on the assumption that granite and log are used for construction which is not guaranteed for each tribe (e.g. the europeans only used processed log for building)